### PR TITLE
Fix issue with long delays when changing enabled filter lists

### PR DESCRIPTION
**Pull request by cuba**
Resolves https://github.com/brave/brave-browser/issues/39283

Caused by issues with the custom text filter list:
- Always tries to load it even if its not set which causes delays to compiling content blockers because its waiting for it to become available
- The search path locatino was incorrect (was applicationDirectory instead of applicationSupportDirectory)

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
See issue for STR

### CODE REVIEW

File: ios/brave-ios/Sources/Brave/WebFilters/CustomFilterListStorage.swift [hunk 1]
```diff
@@ -205,14 +205,20 @@ import WebKit
 extension CustomFilterListStorage {
   /// Gives us source representations of all the enabled custom filter lists
   @MainActor var enabledSources: [GroupedAdBlockEngine.Source] {
-    return
+    var sources =
       filterListsURLs
       .filter(\.setting.isEnabled)
       .map(\.setting.engineSource)
+    if (try? self.savedCustomRulesFileInfo()) == nil { return sources }
```
Comment (soner-yuksel) [inline ios/brave-ios/Sources/Brave/WebFilters/CustomFilterListStorage.swift:212]: I believe savedCustomRulesFileURL check will be sufficient instead

File: ios/brave-ios/Sources/Brave/WebFilters/CustomFilterListStorage.swift [hunk 2]
```diff
@@ -172,7 +172,7 @@ import WebKit
     guard
       let folderURL = FileManager.default.getOrCreateFolder(
         name: "custom_rules",
-        location: .applicationDirectory
+        location: .applicationSupportDirectory
```
Comment (soner-yuksel) [inline ios/brave-ios/Sources/Brave/WebFilters/CustomFilterListStorage.swift:175]: What is the purpose of this change ? Is it the part making it faster ? Using the support directory instead of main directory

File: ios/brave-ios/Sources/Brave/WebFilters/AdBlock/AdBlockGroupsManager.swift [hunk 3]
```diff
@@ -302,7 +302,18 @@ import os
   func compileEngineIfFilesAreReady(for engineType: GroupedAdBlockEngine.EngineType) {
     let manager = self.getManager(for: engineType)
     let enabledSources = sourceProvider.enabledSources(for: engineType)
-    guard manager.checkHasAllInfo(for: enabledSources) else { return }
+    let availableSources = manager.compilableFiles(for: enabledSources)
+      .map({ $0.filterListInfo.source })
+    guard enabledSources.allSatisfy({ availableSources.contains($0) }) else {
```
Comment (soner-yuksel) [inline ios/brave-ios/Sources/Brave/WebFilters/AdBlock/AdBlockGroupsManager.swift:307]: This loop[ is still running even it is not debug and we dont need logs

File: ios/brave-ios/Sources/Brave/WebFilters/CustomFilterListStorage.swift (hunk 2 above)
Comment (cuba) [reply ios/brave-ios/Sources/Brave/WebFilters/CustomFilterListStorage.swift:175]: Basically because using the application directory was a mistake. You can't save files there and the screen has errors and loading the file has errors. This is really the main problem in this PR.

File: ios/brave-ios/Sources/Brave/WebFilters/AdBlock/AdBlockGroupsManager.swift (hunk 3 above)
Comment (cuba) [reply ios/brave-ios/Sources/Brave/WebFilters/AdBlock/AdBlockGroupsManager.swift:307]: I really need logs to test these features out and make sure things are working properly. I missed this mistake and i think i would have spotted it if I had this log. Though I can remove it if you insist.

Comment (cuba) [reply ios/brave-ios/Sources/Brave/WebFilters/AdBlock/AdBlockGroupsManager.swift:307]: Actually just removed it.

**Comment by Uni-verse** [comment]
Verified on `iPhone 12` running `iOS 17.5.1` using version `1.69 (88)`

 - Ensured that error in UI is not shown in Content Filtering.
 - Ensured that adding syntax for Custom Filters is working and verified ads are blocked.
 - Ensured adding and removing filter lists via `Add Filter By URL` is working.
 - Ensured default lists can be enabled without issues. 
 
 settings page | settings added | without filter | with filter
 -|-|-|-
IMG_6690 | IMG_6694 | IMG_6698| IMG_6693
